packages/first-boot-wizard/files/usr/lib/lua/firstbootwizard.lua

```diff
@@ -1,136 +1,107 @@
-        echo -n :
-        echo "$mac" | tail -c +9 | head -c 4
-        echo -n :
-        echo "$mac" | tail -c +13
-    }
-    echo -n `eui64 ]]..mac..'`'
-    return 'fe80::'..execute(cmd)
-end
-
-function file_exists(filename)
-    return fs.stat(filename, "type") == "reg"
-end
-
-local function check_file_exists(file)
-    local f = io.open(file, "rb")
-    if f then f:close() end
-    return f ~= nil
-end  
-
-local function split(str, sep)
-    local sep, fields = sep or ":", {}
-    local pattern = string.format("([^%s]+)", sep)
-    str:gsub(pattern, function(c) fields[#fields+1] = c end)
-    return fields
+-- Write lock file at begin
+function start_scan_file()
+    local file = io.open("/tmp/scanning", "w")
+    file:write("true")
+    file:close()
 end
 
--- splits a multiline string in a list of strings, one per line
-local function lsplit(mlstring)
-    return split(mlstring, "\n")
+-- Remove old results
+function clean_tmp()
+    utils.execute('rm /tmp/lime-defaults__*')
 end
 
-local function phy_to_idx(phy)
-    local substr = string.gsub(phy, "phy", "")
-    return tonumber(substr)
+-- Save working copy of wireless
+function backup_wifi_config()
+    utils.execute("cp /etc/config/wireless /tmp/wireless-temp")
 end
 
+-- Get networks in 5ghz radios
 function get_networks()
-    local thisBssids = {}
-    local wirelessConfig = conn:call("network.wireless", "status", {})
-    local allRadios = wireless.scandevices()
-    -- nixio.syslog("crit", "FBW radios "..json.encode(allRadios))
-    local all_networks = {}
-    local phys = {}
-    for _, radio in pairs (allRadios) do
-        if wireless.is5Ghz(radio[".name"]) then
-            local phyIndex = radio[".name"].sub(radio[".name"], -1)
-            phys[#phys+1] = "phy"..phyIndex
-            -- nixio.syslog("crit", "FBW thisBssids"..json.encode(wirelessConfig["radio"..phyIndex]))
-            table.insert(thisBssids, wirelessConfig["radio"..phyIndex].interfaces[1].config.bssid)
-        end
-    end
-    -- -- nixio.syslog("crit", "FBW thisBssids"..json.encode(thisBssids))
-    -- -- nixio.syslog("crit", "FBW phys"..json.encode(phys))
-    for idx, phy in pairs(phys) do
-        networks = iwinfo.nl80211.scanlist(phy)
-        -- nixio.syslog("crit", "FBW networs"..json.encode(networks))
-        for k,network in pairs(networks) do
-            if network.signal ~= -256 then
-                network["phy"] = phy
-                network["phy_idx"] = phy_to_idx(phy)
-                all_networks[#all_networks+1] = network
-            end
-        end
-    end
-    return all_networks
+    -- Get all radios
+    local radios = ft.map(utils.extract_prop(".name"), wireless.scandevices())
+    -- Get only 5ghz radios
+    local radios_5ghz = ft.filter(wireless.is5Ghz,  radios)
+    -- Convert radios to phys
+    local phys = ft.map(utils.radio_to_phy, radios_5ghz)
+    -- Scan networks in phys and format result
+    local networks = ft.map(
+        function(phy)
```

Comment:
this function looks cryptic here, would recommend putting it outside and comment what it does

Reply:
Changed in favor of **extract_phys_from_radios**